src/main/java/sonarquberepair/DefaultRepair.java

```diff
@@ -98,7 +104,27 @@ public static boolean deleteDirectory(File directoryToBeDeleted) {
 	private Launcher createLauncher(String inputDirPath, String outputDirPath) {
 		Launcher launcher = new Launcher();
 
-		launcher.addInputResource(inputDirPath);
+		if (this.config.getSkipDuplicatedTypes()) { // Load one file at a time , only one per type.
+			Set<String> addedTypes = new HashSet<String>();
+			try {
+				System.out.println("Loading files into spoon launcher ... ");
+				Files.find(Paths.get(inputDirPath),
+						   Integer.MAX_VALUE,
+						   (filePath, fileAttr) -> fileAttr.isRegularFile())
+							.forEach(x -> {
+								if(!addedTypes.contains(x.getFileName().toString())) {
+									addedTypes.add(x.getFileName().toString());
+									launcher.addInputResource(x.toString());
+								}
+							});
+				System.out.println("Done loading.");
+			} catch (IOException e) {
+				launcher.addInputResource(inputDirPath);
+			}
+		} else {
+			launcher.addInputResource(inputDirPath);
+		}
+
```

Comment:
What about `launcher.getEnvironment().setIgnoreDuplicateDeclarations(true);` instead of all those things?

Reply:
Nice . I'll change so that this options will always be true for the launcher.